lib/rule-context.js

```diff
@@ -61,6 +62,71 @@ const PASSTHROUGHS = [
 //------------------------------------------------------------------------------
 
 /**
+ * Compares items in a fixes array by range.
+ * @param {Fix} a The first message.
+ * @param {Fix} b The second message.
+ * @returns {int} -1 if a comes before b, 1 if a comes after b, 0 if equal.
+ * @private
+ */
+function compareFixesByRange(a, b) {
+    return a.range[0] - b.range[0] || a.range[1] - b.range[1];
+}
+
+/**
+ * Merges the given fixes array into one.
+ * @param {Fix[]} fixes The fixes to merge.
+ * @param {SourceCode} sourceCode The source code object to get the text between fixes.
+ * @returns {void}
+ */
+function mergeFixes(fixes, sourceCode) {
```

Comment:
As public API, this is simple enhancement; the `fix` functions support for an iterable object as their return value.

There are 2 topics here.

- To not modify comments, this feature would be helpful.
- To lock ranges, this feature would be helpful.

As I mentioned in this post, currently we have a pain to remain comments. The purpose of the `fix` functions is "to modify code", but we have to focus on "to not modify code" mainly in order to not remove comments. This enhancement would resolve the problem. I think this is valuable for plugin ecosystem as well.

In addition, we can address locked ranges in the same mechanism.

----

In fact, this is the 1st step of my plan.

Because the report of linting has `fix` properties which have a fix object, this PR is merging the multiple fixes before it makes the report for now. I have to avoid a breaking change about the `fix` property (the shape of the report also are public API). 

However, I have a plan (though it's not accepted yet) to change the `fix` property to `fixes` property that has an array of fix objects, so this `mergeFixes` function would be removed in future. It implies that fixing process can distinguish between fixing ranges and locking ranges. I expect it to allow us smarter conflict check.